native/dispatch.c

```diff
@@ -3514,16 +3519,24 @@ Java_com_sun_jna_Native_ffi_1prep_1closure(JNIEnv *env, jclass UNUSED(cls), jlon
   ffi_status s;
 
   if ((*env)->GetJavaVM(env, &cb->vm) != JNI_OK) {
+    free(cb);
     throwByName(env, EUnsatisfiedLink, "Can't get Java VM");
     return 0;
   }
 
   cb->object = (*env)->NewWeakGlobalRef(env, obj);
   cb->closure = ffi_closure_alloc(sizeof(ffi_closure), L2A(&cb->x_closure));
+  if (cb->closure == NULL) {
+    free(cb);
```

Comment:
(Caveat: I am guessing here. Do not consider this review as anything other than seeing a pattern.)

Do we need a `(*env)->DeleteWeakGlobalRef(env, cb->object);` here?

Reply:
Yes I think you're right, see 5299bd5.